### PR TITLE
Adding Flip to JImage

### PR DESCRIPTION
**Pull request by Buddhima**
Hi All,

This PR is to add flip functionality to JImage class, which is using quite lot in Joomla image manipulations.
Sample usage of this method is as follows:

```
$JImage = new JImage($file);

try
{
    $image = $JImage->flip($mode, true);
    $image->toFile($file);

    return true;
}
catch (Exception $e)
{
    $app->enqueueMessage($e->getMessage(), 'error');
}

```

Mode of flipping can have 3 values which are defined in GD library:
- IMG_FLIP_HORIZONTAL
- IMG_FLIP_VERTICAL
- IMG_FLIP_BOTH

Hope this will help for developers to do more with JImage!

Thank you!

### CODE REVIEW

**Comment by C-Lodder** [comment]
Have been trying to test this but constantly getting a blank screen (even with error reporting set to Development).

Wrapped the code in a function and defined the path of an image correctly. Could you please confirm a test you've done with an image that comes shipped with Joomla?

**Comment by Buddhima** [comment]
Hi @C-Lodder ,

Thank you for quick testing. Here's the image results after IMG_FLIP_BOTH mode :

screenshot from 2015-03-24 23 15 20
 
Also feel free to have a look at PR "[#33906] New Media Manager for Joomla! CMS" which has an actual use-case of this.

Thank you!

**Comment by C-Lodder** [comment]
@Buddhima - Oh this is for the Media Editor. Sorry, I was adding your code to a custom module of mine to test

**Comment by Buddhima** [comment]
Hi @C-Lodder ,

This is not just for that. Anyone can use this feature quite similar way as rotate, resize etc.
That's the reason why I pushed this separately.
You can have a look at Media Editor for an example use of this.

Thank you!

**Comment by losedk** [comment]
@test all good. Tested all 3 modes with success

**Comment by Buddhima** [comment]
@losedk , Thank you for testing.

**Comment by C-Lodder** [comment]
@test all good now :)

File: libraries/joomla/image/image.php [hunk 1]
```diff
@@ -845,6 +845,60 @@ public function rotate($angle, $background = -1, $createNew = true)
...
+		{
+			// @codeCoverageIgnoreStart
+			$new = new JImage($handle);
+
+			return $new;
+
+			// @codeCoverageIgnoreEnd
+		}
+		// Swap out the current handle for the new image handle.
+		else
```
Comment (phproberto) [inline libraries/joomla/image/image.php]: Can you please remove this else container as it's not needed? Just leave the contents.

**Comment by Buddhima** [comment]
@phproberto Thanks for pointing that out. Can you check the current code?
Thank you!

**Comment by phproberto** [comment]
@Buddhima it looks good now. Thanks!

**Comment by zero-24** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6571">issues.joomla.org/joomla-cms/6571</a>.</sub>

**Comment by piotr-cz** [comment]
I'd add information on the `$mode` parameter in the docblock.
Either short info or reference to [http://php.net/imageflip#refsect1-function.imageflip-parameters]

**Comment by wilsonge** [comment]
I added a link to the URL you mentioned @piotr-cz during merging into the docblock.

Merged into 3.5 staging. Thanks Buddhima!

**Pull request closed without merging**